### ISSUE TITLE
[Feature][seatunnel-flink-starter] Support read config file from hdfs when run with flink in run-application mode

### ISSUE THREAD

**Issue by zhaoli2333** [apache/seatunnel#5889]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

It's not supported to read a local file when running a flink job using run-application mode. so I uploaded my seatunnel config flie to HDFS in advance, and excuted the command below:
```
/usr/local/seatunnel-2.3.3/bin/start-seatunnel-flink-15-connector-v2.sh --config hdfs://xxx/tmp/config/tidb2kafka --target yarn-application --deploy-mode run-application
```
and got an error like: 
```
SeaTunnelRuntimeException: ErrorCode:[COMMON-01], ErrorDescription:[File operation failed, such as (read,list,write,move,copy,sync) etc...] - Can't find config file hdfs://xxx/tmp/config/tidb2kafka
```

I checked the source code and found seatunnel only support to read a local file as config.

### Usage Scenario

Seatunnel 2.3.3 run with flink engine.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by liugddx** [comment]
@hailin0 @Hisoka-X @EricJoy2048 How about this?

**Comment by Hisoka-X** [comment]
Yes, we should support read config use any protocol, like hdfs, s3, ftp so on. Not only local file.
